client/js/components/main/assessment.jsx

```diff
@@ -30,14 +25,66 @@ const select = (state, props) => {
 export class Assessment extends React.Component{
 
   componentWillMount(){
-    if(this.props.assessmentResult != null){
+    if(this.props.progress.get('assessmentResult') != null){
       appHistory.push("assessment-result");
     }
   }
 
   componentDidMount(){
     // Trigger action to indicate the assessment was viewed
-    this.props.assessmentViewed(this.props.settings, this.props.assessment);
+    this.props.assessmentViewed();
+  }
+
+  /**
+   * Return an item for a given index in props.allQuestions
+   */
+  item(index){
+    let props = this.props;
+    if(props.questionCount === undefined || index >= props.questionCount || index < 0){
+      return <div></div>;
+    }
+    return <Item
+      assessment       = {props.assessment}
+      settings         = {props.settings}
+      question         = {props.allQuestions[index]}
+      currentItemIndex = {index}
+      questionCount    = {props.questionCount}
+      messageIndex     = {-1 /*props.progress.get('answerMessageIndex')*/}
+      allQuestions     = {props.allQuestions}
+      studentAnswers   = {{/*this.props.studentAnswers*/}}
+      confidenceLevels = {{/*this.props.settings.confidence_levels*/}}
+      outcomes         = {props.outcomes}/>;
+  }
+
+  /**
+   * Gets correct number of items for the current view setting
+   */
+  items(){
```

Comment:
This should probably be renamed to item_count(), count_items(), num_items(), or similar.